src/js/components/form/AdvancedSection.js

```diff
@@ -18,6 +18,12 @@ class AdvancedSection extends React.Component {
     });
   }
 
+  componentWillReceiveProps(nextProps) {
```

Comment:
`componentWillReceiveProps` is being deprecated. Could we use [getDerivedStateFromProps](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops) instead?

Reply:
Thanks for the feedback, I will take a look.

Reply:
I couldn't get it working with `getDerivedStateFromProps`. The problem is that the advanced section does not receive the new props, unless something is changed, so we can't close the settings until we click something.

![Peek 2019-04-22 14-51](https://user-images.githubusercontent.com/40791275/56499244-2ec32800-650e-11e9-9d0a-30e55753822d.gif)

`getDerivedStateFromProps` is also not recommended - https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html .

Reply:
I am not sure it's possible to do it without refactoring every component in the chain - `CreateServiceModalForm`, `GeneralServiceFormSection` and `AdvancedSection`.

Reply:
I think I figured it out.

Reply:
@GeorgiSTodorov - If it's a lot of work to avoid using `componentWillReceiveProps` here, I'll still approve because it's not deprecated until React 17.

I'll check back tomorrow to see if you pushed up a change :)

Reply:
I will try some more, but it's definitely tricky. If I don't do it today, I will keep it in mind for my next spike day.